src/scenes/App/components/TermsAndConditions/index.jsx

```diff
@@ -0,0 +1,274 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { withStyles } from 'material-ui/styles';
+import Paper from 'material-ui/Paper';
+import Typography from 'material-ui/Typography';
+import { FormattedMessage } from 'react-intl';
+import Grid from 'material-ui/Grid';
+import { FormGroup, FormControlLabel } from 'material-ui/Form';
+import Checkbox from 'material-ui/Checkbox';
+import Button from 'material-ui/Button';
+
+import AppConfig from '../../../../config/app';
+import styles from './styles';
+
+const TNC_ACCEPTED = 'tncAccepted';
+
+@withStyles(styles, { withTheme: true })
+export default class TermsAndConditions extends React.PureComponent {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    locale: PropTypes.string.isRequired,
+    langHandler: PropTypes.func,
+  };
+
+  static defaultProps = {
+    langHandler: undefined,
+  };
+
+  state = {
+    checkboxChecked: false,
+  };
+
+  render() {
+    const { classes, locale } = this.props;
+    const { checkboxChecked } = this.state;
+
+    let tncAccepted = localStorage.getItem(TNC_ACCEPTED);
+    tncAccepted = _.isUndefined(tncAccepted) || _.isNull(tncAccepted) ? false : tncAccepted;
+
+    return (
+      <div
+        className={classes.tncBg}
+        style={
+          {
+            opacity: tncAccepted ? 0 : 1,
+            display: tncAccepted ? 'none' : 'block',
+          }
+        }
+      >
+        <Grid container>
+          <Grid item xs>
+            <Button color="primary" variant="raised" className={classes.langBtn} onClick={this.props.langHandler}>
+              <FormattedMessage id="language.select" defaultMessage="中文" />
+            </Button>
+          </Grid>
+          <Grid item xs={8}>
+            <Paper className={classes.tncWrapper}>
+              {this.renderTitle()}
```

Comment:
For future reference, `renderXXX` is bad practice [see why here](https://github.com/airbnb/javascript/issues/520#issuecomment-306196763) and also, it's in the [react best practices guide](https://github.com/alex-cory/fasthacks/blob/master/notes/react/best-practices.md) I made.  Please change to `<Title />`, `<General />` , etc.

Reply:
done